plugins/module_utils/network/ios/facts/acls/acls.py

```diff
@@ -132,20 +132,18 @@ def populate_facts(self, connection, ansible_facts, data=None):
 
             def factor_source_dest(ace, typ):
                 temp = ace.get(typ, {})
+                # TODO add the XOR logic for fixing ipv6 detla address issue
                 if temp.get("address"):
-                    _temp_addr = temp.get("address", "")
-                    ace[typ]["address"] = _temp_addr.split(" ")[0]
-                    ace[typ]["wildcard_bits"] = _temp_addr.split(" ")[1]
-                if temp.get("ipv6_address"):
-                    _temp_addr = temp.get("ipv6_address", "")
-                    if len(_temp_addr.split(" ")) == 2:
-                        ipv6_add = ace[typ].pop("ipv6_address")
-                        ace[typ]["address"] = ipv6_add.split(" ")[0]
-                        ace[typ]["wildcard_bits"] = ipv6_add.split(" ")[1]
-                    else:
-                        ace[typ]["address"] = ace[typ].pop("ipv6_address")
+                    ace[typ]["address"] = temp.get("address")
+                    ace[typ]["wildcard_bits"] = temp.get("wildcard_bits")
 
             def process_protocol_options(each):
+                ICMP_MAP = {
```

Comment:
We are handling only one?
https://github.com/ansible-collections/cisco.ios/blob/main/plugins/module_utils/network/ios/config/acls/acls.py#L392 Can this be used instead?